src/robusta/core/triggers/oom_killed_trigger_base.py

```diff
@@ -0,0 +1,86 @@
+from typing import List
+from abc import abstractmethod
+
+from hikaru.model import Pod
+
+from ..model.pods import pod_most_recent_oom_killed_container
+from ...core.playbooks.base_trigger import TriggerEvent
+from ...integrations.kubernetes.autogenerated.triggers import (
+    PodUpdateTrigger,
+    PodChangeEvent,
+)
+from ...integrations.kubernetes.base_triggers import K8sTriggerEvent
+from ...utils.rate_limiter import RateLimiter
+from pydantic import BaseModel
+
+
+class Exclude(BaseModel):
+    name: str = None
+    namespace: str = None
+
+
+class OOMKilledTriggerBase(PodUpdateTrigger):
+    """
+    :var rate_limit: Limit firing to once every `rate_limit` seconds
+    """
+
+    rate_limit: int = 0
+    exclude: List[Exclude] = None
+
+    def __init__(
+            self,
+            name_prefix: str = None,
+            namespace_prefix: str = None,
+            labels_selector: str = None,
+            rate_limit: int = 0,
+            exclude: List[Exclude] = None,
+    ):
+        super().__init__(
+            name_prefix=name_prefix,
+            namespace_prefix=namespace_prefix,
+            labels_selector=labels_selector,
+        )
+        self.rate_limit = rate_limit
+        self.exclude = exclude
+
+    @abstractmethod
+    def get_relevant_oomkilled_containers(self, pod: Pod):
+        return []
+
+    def should_fire(self, event: TriggerEvent, playbook_id: str):
+        should_fire = super().should_fire(event, playbook_id)
+        if not should_fire:
+            return should_fire
+
+        if not isinstance(event, K8sTriggerEvent):
+            return False
+
+        exec_event = self.build_execution_event(event, {})
+
+        if not isinstance(exec_event, PodChangeEvent):
+            return False
+
+        pod = exec_event.get_pod()
+
+        containers = self.get_relevant_oomkilled_containers(pod)
+
+        if len(containers) == 0:
+            return False
+
+        oom_killed = pod_most_recent_oom_killed_container(pod,containers=containers, only_current_state=True)
```

Comment:
When do you use only_current_state?

Reply:
everywhere else specifically in the actions, 
We have seen a difference in state at the time the trigger sees the states and the time an action gets to see the states
for our example if we run three actions, by the time some of the actions are run the current state can already be "running" again and the last state will be "OOMKilled"
This was seen allot in testing